geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -160,40 +152,44 @@ public int getDSFID() {
     return REDIS_SORTED_SET_ID;
   }
 
-  protected synchronized byte[] memberAdd(byte[] memberToAdd, byte[] scoreToAdd) {
-    OrderedSetEntry existingEntry = members.get(memberToAdd);
-    if (existingEntry == null) {
+  protected synchronized boolean memberAdd(byte[] memberToAdd, double scoreToAdd) {
+    OrderedSetEntry entry = members.get(memberToAdd);
+    if (entry == null) {
       OrderedSetEntry newEntry = new OrderedSetEntry(memberToAdd, scoreToAdd);
       members.put(memberToAdd, newEntry);
       scoreSet.add(newEntry);
-      return null;
+      return false;
     } else {
-      scoreSet.remove(existingEntry);
-      byte[] oldScore = existingEntry.scoreBytes;
-      existingEntry.updateScore(stripTrailingZeroFromDouble(scoreToAdd));
-      members.put(memberToAdd, existingEntry);
-      scoreSet.add(existingEntry);
-      return oldScore;
+      if (entry.score == scoreToAdd) {
+        return false;
+      }
+      updateScore(memberToAdd, scoreToAdd, entry);
     }
+    return true;
+  }
+
+  private void updateScore(byte[] memberToAdd, double scoreToAdd, OrderedSetEntry entry) {
+    scoreSet.remove(entry);
+    entry.updateScore(scoreToAdd);
+    members.put(memberToAdd, entry);
```

Comment:
I think this line is not necessary since this is a reference to the value already in the `members` map. Could you also rather inline the remaining 3 lines since this method isn't used anywhere else.